apps/api/src/routers/admin.py

```diff
@@ -558,14 +558,20 @@ async def subevent_checkin(
     await participant_manager.subevent_checkin(event, uid, organizer)
 
 
-async def _process_records_in_batches(records: list[dict[str, object]]) -> None:
-    for decision in (Decision.ACCEPTED, Decision.WAITLISTED, Decision.REJECTED):
-        group = [record for record in records if record["decision"] == decision]
-        if not group:
-            continue
-        await asyncio.gather(
-            *(_process_batch(batch, decision) for batch in batched(group, 100))
-        )
+async def _process_records_in_batches(
+    records: list[dict[str, object]], application_types: list[Role]
+) -> None:
+    for application_type in application_types:
```

Comment:
Suggestion: rather than iterating here, the consumer could just call this function multiple times.